dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/managegroups/ManageGroupsCRUDServiceIT.java

```diff
@@ -0,0 +1,122 @@
+    @EJB
+    private ManageGroupsCRUDService manageGroupsCRUDService;
+    @EJB
+    private DataverseServiceBean dataverseService;
+    @Inject
+    private DataverseSession dataverseSession;
+    @EJB
+    private AuthenticationServiceBean authenticationService;
+    @EJB
+    private ExplicitGroupServiceBean explicitGroupService;
+
+    @Before
+    public void setUp() {
+        dataverseSession.setUser(authenticationService.getAdminUser());
+    }
+
+    @Test
+    public void shouldCreateGroup() {
+        // given
+        Dataverse dv = dataverseService.findByAlias("ownmetadatablocks");
+        RoleAssignee roleAssignee = authenticationService.findAllAuthenticatedUsers().get(0);
+
+        // when
+        manageGroupsCRUDService.create(dv, "testGroup", "testGroupId", "testDesc", Collections.singletonList(roleAssignee));
+
+        // then
+        List<ExplicitGroup> dbExplicitGroupList = explicitGroupService.findByOwner(dv.getId());
+        Assert.assertTrue(dbExplicitGroupList
+                .stream()
+                .anyMatch(explicitGroup -> explicitGroup.getDisplayName().equals("testGroup")));
+        Assert.assertTrue(dbExplicitGroupList
+                .stream()
+                .filter(explicitGroup -> explicitGroup.getDisplayName().equals("testGroup"))
+                .findFirst().get().structuralContains(roleAssignee));
+    }
+
+    @Test
+    public void shouldUpdateGroup() {
+        // given
+        Dataverse dv = dataverseService.findByAlias("ownmetadatablocks");
+        ExplicitGroup explicitGroup = getExplicitGroup(dv);
+
+        em.persist(explicitGroup);
+
+        // when
+        explicitGroup.setDisplayName("testGroup");
+        manageGroupsCRUDService.update(explicitGroup, new LinkedList<>());
+
+        // then
+        List<ExplicitGroup> dbExplicitGroupList = explicitGroupService.findByOwner(dv.getId());
+        Assert.assertTrue(dbExplicitGroupList.stream().anyMatch(eg -> eg.getId().equals(explicitGroup.getId())));
+        Assert.assertTrue(dbExplicitGroupList.stream().anyMatch(eg -> eg.getDisplayName().equals("testGroup")));
+        Assert.assertFalse(dbExplicitGroupList.stream().anyMatch(eg -> eg.getDisplayName().equals("explicitGroupName")));
+    }
+
+    @Test
+    public void shouldDeleteGroup() {
+        // given
+        Dataverse dv = dataverseService.findByAlias("ownmetadatablocks");
+        ExplicitGroup explicitGroup = getExplicitGroup(dv);
+
+        em.persist(explicitGroup);
+
+        // when
+        manageGroupsCRUDService.delete(explicitGroup);
+
+        // then
+        List<ExplicitGroup> dbExplicitGroupList = explicitGroupService.findByOwner(dv.getId());
+        Assert.assertFalse(dbExplicitGroupList.stream().anyMatch(eg -> eg.getId().equals(explicitGroup.getId())));
+        Assert.assertFalse(dbExplicitGroupList.stream().anyMatch(eg -> eg.getDisplayName().equals("explicitGroupName")));
+    }
+
+    // -------------------- PRIVATE ---------------------
+    @NotNull
```

Comment:
What this annotation does?

Reply:
Must have been added by IntelliJ when auto-creating private method for selected junk of code. 
Dunno what it does :)

Reply:
Well. If you don't know what it does then why it is here?

Reply:
Sorry, forgot to remove this in last pr.